train.py

```diff
@@ -182,7 +182,7 @@
 best_val_epoch = 0
 training_step = 0
 num_validations = 0
-dataloader_length = train_df.shape[0]
+dataloader_length = int(train_df.shape[0] / batch_size)
```

Comment:
I don't think this is generally correct since `int(0.5) == 0`, but you can just use `len(train_dataloader)`

Reply:
I figured the rounding error would be minimal. But for sure `len(dataloader)` is better - I had googled this and found something which implied it wasn't possible, but it must have been wrong since `DataLoader` implements `__len__`.

Reply:
It probably is 😄 I think `len(dataloader)` can have small inaccuracies when you have an iterable dataset (no `__getitem__`) and/or with multiprocessing but in our case I think this is accurate and easiest (it takes into account `drop_last=...` etc.)